ARM-wvd-templates/nestedtemplates/managedDisks-customimagevm.json

```diff
@@ -1,387 +1,473 @@
-                    "user": "[parameters('administratorAccountUsername')]",
-                    "restart": "true",
-                    "options": "3"
-                },
-                "protectedSettings": {
-                    "password": "[parameters('administratorAccountPassword')]"
-                }
+          },
+          "imageReference": {
+            "id": "[parameters('rdshImageSourceId')]"
+          }
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '-nic'))]"
             }
+          ]
         },
-        {
-            "apiVersion": "2018-10-01",
-            "type": "Microsoft.Compute/virtualMachines/extensions",
-            "name": "[concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '/', 'dscextension')]",
-            "location": "[parameters('location')]",
-            "dependsOn": ["rdsh-domain-join-loop"],
-            "copy": {
-                "name": "rdsh-dsc-loop",
-                "count": "[parameters('rdshNumberOfInstances')]"
-            },
-            "properties": {
-                "publisher": "Microsoft.Powershell",
-                "type": "DSC",
-                "typeHandlerVersion": "2.73",
-                "autoUpgradeMinorVersion": true,
-                "settings": {
-                    "modulesUrl": "[parameters('artifactsLocation')]",
-                    "configurationFunction": "Configuration.ps1\\AddSessionHost",
-                    "properties": {
-                        "hostPoolName": "[parameters('hostpoolName')]",
-                        "registrationInfoToken": "[parameters('hostpoolToken')]"
-                    }
-                }
-            }
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": false
+          }
+        },
+        "licenseType": "Windows_Client"
+      },
+      "zones": "[if(equals(parameters('availabilityOption'), 'AvailabilityZone'), array(parameters('availabilityZone')), variables('emptyArray'))]"
+    },
+    {
+      "apiVersion": "2018-10-01",
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '/', 'Microsoft.PowerShell.DSC')]",
+      "location": "[parameters('location')]",
+      "dependsOn": [ "rdsh-vm-loop" ],
+      "copy": {
+        "name": "rdsh-dsc-loop",
+        "count": "[parameters('rdshNumberOfInstances')]"
+      },
+      "properties": {
+        "publisher": "Microsoft.Powershell",
+        "type": "DSC",
+        "typeHandlerVersion": "2.73",
+        "autoUpgradeMinorVersion": true,
+        "settings": {
+          "modulesUrl": "[parameters('artifactsLocation')]",
+          "configurationFunction": "Configuration.ps1\\AddSessionHost",

```

Comment:
Does the code this DSC task runs require the VM to be domain joined? This code changes the DSC code to run before the domain join.

Reply:
I just sent you an email with some info abou this.

Reply:
AADJ requires DSC code to run before domain join

Reply:
Scott is emailing to double check on the non-AADJ case.

Reply:
Confirmation for non-AADJ case by Scott's email, good to go.